### ISSUE TITLE
Windows Forms locks on local async data portal call

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/csla#1164]
As per @michaelcsikos in a [forum thread](https://github.com/MarimerLLC/cslaforum/issues/753).

**Describe the bug**
I noticed that my UI is blocking even though I am using the `async` methods in the `DataPortal`. I'm running the `DataPortal` on the client side. Up until 4.6.603, the `async` methods correctly ran in a non-blocking thread. With 4.10.1, they're blocking. Did I miss a memo?

I have attached two versions of the same project which simply check the `Thread.CurrentThread.ManagedThreadId`. In 4.6.603, the thread Id inside `DataPortal_Create` is a new thread, but in 4.10.1, it's running on the UI thread.

**Version and Platform**
CSLA version: 4.10.1 (and 5.0)
OS: Windows
Platform: WinForms

[CslaAsync.zip](https://github.com/MarimerLLC/cslaforum/files/3227305/CslaAsync.zip)

**Comment by rockfordlhotka** [comment]
I suspect it has something to do with this change in February:

https://github.com/MarimerLLC/csla/commit/401c0a266ba37348fc2f9a16c2daeef52e95cf54#diff-f11955a5e47bc82ce5ab25d671dcb06f

PR: https://github.com/MarimerLLC/csla/pull/1073

So we solved an issue in ASP.NET, but sadly appear to have created an issue in WinForms (and probably WPF) 😞

**Comment by rockfordlhotka** [comment]
I've been thinking about this a bit. Although the Feb 2019 change by @kellyethridge "fixed" an issue with flowing HttpContext to worker threads, I am left wondering if it didn't create a different issue even in ASP.NET?

Specifically, what the change appears to have done is caused worker threads on the thread pool to run within the same sync context (obviously) - and (not so obviously) this apparently means that those "worker threads" are actually the UI thread - so it is blocking the use of background threads entirely. At least in Windows Forms - so I then wonder if it is doing the same in ASP.NET?

In other words, did this change block the use of _actual_ background threads entirely? Or only in WinForms (and probably WPF)?

**Comment by michaelcsikos** [comment]
Defaulting to the old behaviour with an optional config for ASP.NET seems reasonable to me.

We've embraced `async` wholeheartedly in our applications, which means we've got a couple hundred users out there who are probably wondering why our recent versions feel so slow. As a developer, most of my testing is done with a local SQL Server database, so I didn't really notice the change until a couple of days ago. Really looking forward to 5.0.0 :smiley:

**Comment by michaelcsikos** [comment]
Hi @rockfordlhotka 

I'm just wondering if this could be released in a 4.10.x update, as I see 5.0.0 is still a fair way off. I think it may have side effects which are more serious than just blocking the UI, as I mentioned [here](https://github.com/MarimerLLC/cslaforum/issues/759).

I'm also very keen to get [Reorder enum values so compiler directive has no x-plat effect](https://github.com/MarimerLLC/csla/pull/1143).

**Comment by rockfordlhotka** [comment]
I expect to release 4.11.0 with the fix tomorrow.

I think the enum value fix is in there too?

**Comment by rockfordlhotka** [comment]
@michaelcsikos sadly this backport wasn't in 4.11.0, but is now in 4.11.1

**Comment by michaelcsikos** [comment]
No worries, thanks, Rocky.
